addons/mass_mailing/static/src/js/mass_mailing_widget.js

```diff
@@ -373,8 +379,15 @@ var MassMailingFieldHtml = FieldHtml.extend({
         if (!odoo.debug) {
             $snippetsSideBar.find('.o_codeview_btn').hide();
         }
-        const $codeview = this.$content.parent().find('textarea.o_codeview');
-        $snippetsSideBar.on('click', '.o_codeview_btn', () => this._toggleCodeView($codeview));
+        const $codeview = this.$content.parent().parent().find('textarea.o_codeview');
+        $snippetsSideBar.on('click', '.o_codeview_btn', () => {
+            this._toggleCodeView($codeview);
+            const isFullscreen = !!$(window.top.document).find('.o_mass_mailing_mailing_form_nosheet')[0];
+            $codeview.css('height', isFullscreen ? $(window).height() : '');
+            if ($codeview.hasClass('d-none')) {
+                core.bus.trigger('iframe_updated', this.wysiwyg.$iframe);
+            }
```

Comment:
Can't this be part of a "_toggleCodeView" method override ?

Reply:
Yup.